pkg/controller/scyllacluster/resource.go

```diff
@@ -1258,3 +1319,284 @@ func MakeJobs(sc *scyllav1.ScyllaCluster, services map[string]*corev1.Service, i
 
 	return jobs, progressingConditions
 }
+
+func MakeEndpointSlicesAndEndpoints(sc *scyllav1.ScyllaCluster, services map[string]*corev1.Service, podLister corev1listers.PodLister) ([]*discoveryv1.EndpointSlice, []*corev1.Endpoints, error) {
+	var endpointSlices []*discoveryv1.EndpointSlice
+	var endpoints []*corev1.Endpoints
```

Comment:
I think one should be a projection of the other, so i might be a dedicated function to build one and then convert it